### PR TITLE
chore: Setup CI with GitHub Actions

### PR DESCRIPTION
**Pull request by sawood14012**

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
```
Comment (arajkumar) [inline .github/workflows/ci-build.yml:48]: This is not required. go test do the magic.

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (sawood14012) [reply .github/workflows/ci-build.yml:48]: okay will remove

Comment (sawood14012) [reply .github/workflows/ci-build.yml]: done

**Review by arajkumar — APPROVED** [review]
lgtm

**Pull request merged**